triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -101,9 +110,12 @@ class Server(
       .asInstanceOf[Option[ActorRef[TriggerRunner.Message]]]
 
   private def startTrigger(
-      credentials: UserCredentials,
+      userpass: (String, String),
       triggerName: Identifier,
       existingInstance: Option[UUID] = None): Either[String, JsValue] = {
+    val credentials = UserCredentials(encrypt(secretKey, userpass._1, userpass._2))
+    val ledgerAccessToken: Option[LedgerAccessToken] = authServiceClient.map(client =>
+      Await.result(client.getLedgerToken(userpass._1, userpass._2, ""), 10.seconds))
```

Comment:
`Await.result` is usually a bad idea, can we turn `startTrigger` into a future?

Reply:
If the authentication fails we want the `start` request to fail synchronously, right?

Reply:
We do but that doesn’t stop you from using a future here. The `upload_dar` endpoint also makes use of a future to stream the request body so that might be a useful example. The problem with `Await.result` is that it blocks the whole thread. The main reason for using futures is that it allows you to dispatch lots of things on a small number of threads.

Reply:
Ah I see. Ok I'll try to use a similar approach